From e6875e1eb393b662a29ac94fdc601d02a920677b Mon Sep 17 00:00:00 2001
Message-ID: <e6875e1eb393b662a29ac94fdc601d02a920677b.1701297273.git.sharlatanus@gmail.com>
In-Reply-To: <123a0b9b76c24c29184c527dabb652384f7be762.1701297273.git.sharlatanus@gmail.com>
References: <123a0b9b76c24c29184c527dabb652384f7be762.1701297273.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sat, 28 Oct 2023 03:16:25 +0100
Subject: [PATCH v3 08/66] gnu: go-github-com-gorilla-websocket: Move to (gnu
 packages golang-web).

* gnu/packages/golang.scm (go-github-com-gorilla-websocket): Move from
here...
* gnu/packages/golang-web.scm: ...to here.

* gnu/packages/ipfs.scm: Add (gnu packages golang-web) module.

Change-Id: I844738ec0b178a6a1746b01659c93a20919c2736
---
 gnu/packages/golang-web.scm | 23 +++++++++++++++++++++++
 gnu/packages/golang.scm     | 21 ---------------------
 gnu/packages/ipfs.scm       |  1 +
 3 files changed, 24 insertions(+), 21 deletions(-)

diff --git a/gnu/packages/golang-web.scm b/gnu/packages/golang-web.scm
index 79a69bc842..0ca70cec38 100644
--- a/gnu/packages/golang-web.scm
+++ b/gnu/packages/golang-web.scm
@@ -5,6 +5,7 @@
 ;;; Copyright © 2021, 2022 Efraim Flashner <efraim@flashner.co.il>
 ;;; Copyright © 2021 Sarah Morgensen <iskarian@mgsn.dev>
 ;;; Copyright © 2022 Giacomo Leidi <goodoldpaul@autistici.org>
+;;; Copyright © 2022 muradm <mail@muradm.net>
 ;;;
 ;;; This file is part of GNU Guix.
 ;;;
@@ -161,6 +162,28 @@ (define-public go-github-com-gorilla-mux
 incoming requests with their respective handler.")
     (license license:bsd-3)))
 
+(define-public go-github-com-gorilla-websocket
+  (package
+    (name "go-github-com-gorilla-websocket")
+    (version "1.5.0")
+    (home-page "https://github.com/gorilla/websocket")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url home-page)
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "1xrr6snvs9g1nzxxg05w4i4pq6k1xjljl5mvavd838qc468n118i"))))
+    (build-system go-build-system)
+    (arguments
+     `(#:import-path "github.com/gorilla/websocket"))
+    (synopsis "Fast WebSocket implementation for Go")
+    (description "Gorilla WebSocket is a Go implementation of the WebSocket
+protocol.")
+    (license license:bsd-2)))
+
 (define-public go-github-com-puerkitobio-goquery
   (package
     (name "go-github-com-puerkitobio-goquery")
diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index a4f13b8c9d..a3c6df391b 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -12773,27 +12773,6 @@ (define-public go-github-com-google-shlex
 using shell-style rules for quoting and commenting.")
     (license license:asl2.0)))
 
-(define-public go-github-com-gorilla-websocket
-  (package
-    (name "go-github-com-gorilla-websocket")
-    (version "1.5.0")
-    (home-page "https://github.com/gorilla/websocket")
-    (source (origin
-              (method git-fetch)
-              (uri (git-reference
-                    (url home-page)
-                    (commit (string-append "v" version))))
-              (file-name (git-file-name name version))
-              (sha256
-               (base32
-                "1xrr6snvs9g1nzxxg05w4i4pq6k1xjljl5mvavd838qc468n118i"))))
-    (build-system go-build-system)
-    (arguments
-     `(#:import-path "github.com/gorilla/websocket"))
-    (synopsis "Fast WebSocket implementation for Go")
-    (description "Gorilla WebSocket is a Go implementation of the WebSocket protocol.")
-    (license license:bsd-2)))
-
 (define-public go-github-com-sourcegraph-jsonrpc2
   (package
     (name "go-github-com-sourcegraph-jsonrpc2")
diff --git a/gnu/packages/ipfs.scm b/gnu/packages/ipfs.scm
index a4c0613e40..fc66582c8a 100644
--- a/gnu/packages/ipfs.scm
+++ b/gnu/packages/ipfs.scm
@@ -30,6 +30,7 @@ (define-module (gnu packages ipfs)
   #:use-module (guix build-system go)
   #:use-module (gnu packages golang)
   #:use-module (gnu packages golang-check)
+  #:use-module (gnu packages golang-web)
   #:use-module (gnu packages python)
   #:use-module (gnu packages shells)
   #:use-module (gnu packages syncthing))
-- 
2.41.0

